### PR TITLE
Api view

### PR DESCRIPTION
**Pull request by CannonLock**
I will wait to merge until #268 is merged and then drop my final commit on that.

### CODE REVIEW

**Comment by bbockelm** [comment]
@CannonLock - I think this is ready for a rebase now that #268 is in.

**Comment by bbockelm** [comment]
Also -- before merging, can you writeup a corresponding issue?  I don't see any obvious one that this is associated with.

**Comment by CannonLock** [comment]
Close #251

**Comment by haoming29** [comment]
For some reason when I navigate to `/view/config` the page is blank, nothing is under `<body>` tag, no console error, can I can see `/view/config` returns 200 at backend. @CannonLock Do you think I missed anything in setting up?

File: origin_ui/src/app/(dashboard)/config/page.tsx [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+import TextField from "@mui/material/TextField";
+import {ArrowDropDown, ArrowDropUp} from '@mui/icons-material';
+import {fontSize} from "@mui/system";
+
+interface Config {
+    ConfigDir: string;
+    Debug: boolean;
+    TLSSkipVerify: boolean;
+    IssuerKey: string;
+    Transport: {
```
Comment (haoming29) [inline origin_ui/src/app/(dashboard)/config/page.tsx:40]: I think all the types under `Transport` are number not string

File: origin_ui/src/app/(dashboard)/config/page.tsx [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+            disabled
+            size="small"
+            id={`${id}-text-input`}
+            label={name}
+            variant={"outlined"}
+            value={value}
+        />
+    }
+
+    if(typeof value === 'number' || value instanceof Number){
```
Comment (haoming29) [inline origin_ui/src/app/(dashboard)/config/page.tsx:163]: Is it possible that we add unit for the values under `Transport`? Right now having 10 digit number as values for various "duration" fields can confuse people, or we might want to convert the durations into second when displaying them (I can do the conversion at the backend if that works better)

Screenshot 2023-11-02 at 9 07 57 AM

File: origin_ui/src/app/globals.css
Comment (haoming29) [inline origin_ui/src/app/globals.css:1]: We might also want to change the tick color of the graph to be lighter. It's hard to see it in dark mode right now.

Ref:
Screenshot 2023-11-02 at 9 20 39 AM

File: origin_ui/src/app/(dashboard)/Sidebar.tsx [hunk 3]
```diff
@@ -17,25 +17,44 @@
...
+                    </Box>
+                    <Box pt={1}>
+                        <Link href={"/config/"}>
+                            <IconButton>
+                                <BuildIcon/>
+                            </IconButton>
+                        </Link>
+                    </Box>
                 </div>
                 <div>
```
Comment (haoming29) [inline origin_ui/src/app/(dashboard)/Sidebar.tsx:59]: It seems that GitHub icon is not aligned with others in the column. Not sure if it's an existing bug or not.

Ref:
Screenshot 2023-11-02 at 9 28 17 AM

**Review by haoming29 — CHANGES_REQUESTED** [review]
Thanks for taking care of the dark mode! It looks fantastic and very cool. A couple of small changes requested plus the `/config/` page not showing up when navigated from index through the sidebar. 

I'm not entirely sure why we would want to check if there's a trailing slash before feeding the next.js `index.html`? I thought it's more common not to append the slash to the end of the page url.

File: origin_ui/src/app/(dashboard)/Sidebar.tsx [hunk 4]
```diff
@@ -17,25 +17,44 @@
...
-                <div style={{display:"flex"}}>
-                    <Image
-                        src={PelicanLogo}
-                        alt={"Pelican Logo"}
-                        width={32}
-                        height={32}
-                    />
+            <div className={styles.header} style={{display: "flex", flexDirection: "column", justifyContent:"space-between", padding:"1rem", top:0, position:"fixed", zIndex:"1", overflow: "hidden", height: "100vh"}}>
+                <div style={{display:"flex", flexDirection: "column"}}>
+                    <Link href={"/index.html"}>
```
Comment (haoming29) [inline origin_ui/src/app/(dashboard)/Sidebar.tsx:36]: This doesn't seem to be the correct way to patch the issue. Why do we need to append the `index.html` to call the URLs? Isn't this supposed to be handled by the web server (either Go or Nginx)? For running the origin without Nginx, Go should be able to direct all page traffic to`index.html`

Comment (CannonLock) [reply origin_ui/src/app/(dashboard)/Sidebar.tsx:36]: I think we agreed the although it isn't great that we have to be explicit in our request this is a fix.

File: origin_ui/src/app/globals.css
Comment (CannonLock) [reply origin_ui/src/app/globals.css:1]: I double checked and I have removed all the dark mode code. Do a hard reload to clear out the existing stylesheets.

**Comment by CannonLock** [comment]
@haoming29 Lets move the enhanced view of duration type objects to the next milestone. #322

**Review by haoming29 — APPROVED** [review]
LGTM! But let's resolve the merge conflict against main first. I don't want to mess up with the discrepancy in `package.json` and @CannonLock you should have the best knowledge in resolving the conflicts.

**Pull request merged**